src/replication/log_consumer.hpp

```diff
@@ -163,6 +148,17 @@ namespace cubreplication
       void set_stop (void);
   };
 
+  //
+  // dispatch_consumer is the common interface used by dispatch to control group creation.
+  //
+  class dispatch_consumer
```

Comment:
The interface does not look generic enough to be used by other class than slave's GCM.

What is the advantage of using a pointer to virtual interface instead of using slave GCM instance directly?